internal/dcache/metadata_manager/metadata_manager_impl.go

```diff
@@ -34,62 +34,223 @@
-	// Dummy implementation
-	return nil
+	err := m.storageCallbacks.DeleteBlobInStorage(internal.DeleteFileOptions{
+		Name: filepath.Join(m.cacheDir, "Objects", filePath),
+	})
+	if err != nil {
+		if bloberror.HasCode(err, bloberror.BlobNotFound) {
+			log.Warn("DeleteFile :: DeleteBlobInStorage failed since blob is already deleted")
+			return nil
+		}
+	}
+	return err
 }
 
 // OpenFile increments the open count for a file and returns the updated count
 func (m *BlobMetadataManager) OpenFile(filePath string) (int64, error) {
-	// Dummy implementation
-	return 0, nil
+	count, err := m.updateHandleCount(filePath, true)
+	if err != nil {
+		log.Debug("OpenFile :: Failed to update file open count: %v", err)
+		return -1, err
+	}
+	return count, nil
 }
 
 // CloseFile decrements the open count for a file and returns the updated count
 func (m *BlobMetadataManager) CloseFile(filePath string) (int64, error) {
-	// Dummy implementation
-	return 0, nil
+	count, err := m.updateHandleCount(filePath, false)
+	if err != nil {
+		log.Debug("CloseFile :: Failed to update file close count: %v", err)
+		return -1, err
+	}
+	return count, nil
+}
+
+func (m *BlobMetadataManager) updateHandleCount(filePath string, increment bool) (int64, error) {
+	const maxBackoff = 30           // Maximum backoff time in seconds
+	backoff := 1 * time.Millisecond // Initial backoff time in milliseconds
+	var openCount int
+	for {
+		// Get the current handle count
+		attr, err := m.storageCallbacks.GetPropertiesFromStorage(internal.GetAttrOptions{
+			Name: filepath.Join(m.cacheDir, "Objects", filePath),
+		})
+		if err != nil {
+			log.Debug("updateHandleCount :: Failed to get handle count: %v", err)
+			return -1, err
+		}
+
+		if attr.Metadata["opencount"] != nil {
+			openCount, err = strconv.Atoi(*attr.Metadata["opencount"])
+			if err != nil {
+				log.Debug("GetFileOpenCount :: Failed to parse handle count: %v", err)
+				return -1, err
+			}
+			if increment {
+				openCount++
+			} else {
+				openCount--
+			}
+			if openCount < 0 {
+				openCount = 0
+			}
+			openCountStr := strconv.Itoa(openCount)
+			attr.Metadata["opencount"] = &openCountStr
+		}
+
+		// Set the new metadata in storage
+		err = m.storageCallbacks.SetMetaPropertiesInStorage(internal.SetMetadataOptions{
+			Path:      filepath.Join(m.cacheDir, filePath),
+			Metadata:  attr.Metadata,
+			Etag:      to.Ptr(azcore.ETag(attr.ETag)),
+			Overwrite: true,
+		})
+		if err != nil {
+			if bloberror.HasCode(err, bloberror.ConditionNotMet) {
+				log.Warn("updateHandleCount :: SetPropertiesInStorage failed due to ETag mismatch, retrying...")
+
+				// Apply exponential backoff
+				log.Debug("updateHandleCount :: Retrying in %d milliseconds...", backoff)
+				time.Sleep(time.Duration(backoff) * time.Millisecond)
+
+				// Double the backoff time, but cap it at maxBackoff
+				backoff *= 2
+				if backoff > maxBackoff {
+					backoff = maxBackoff
+				}
```

Comment:
we should have some exit condition too, to avoid indefinite wait.
Retrying till 1 minute and then exiting is probably safe